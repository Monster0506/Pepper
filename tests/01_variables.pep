%% Test: Variable Declaration and Basic Operations

%% Integer variables
LET x: int = 10 %% assign 10 to x
LET y: int = 5
SHOW(x)
SHOW(y)

%% Float variables
LET pi: float = 3.14159
LET e: float = 2.71828
SHOW(pi)
SHOW(e)

%% String variables
LET greeting: string = "Hello"
LET name: string = "World"
SHOW(greeting)
SHOW(name)

%% Boolean variables
LET isTrue: bool = true
LET isFalse: bool = false
SHOW(isTrue)
SHOW(isFalse)

%% Basic arithmetic
REAS x = x y +
SHOW(x)
REAS y = x y *
SHOW(y)
REAS pi = pi 2 *
SHOW(pi)
