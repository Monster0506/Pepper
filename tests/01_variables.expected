10
5
3.14159
2.71828
Hello
World
true
false
15
75
6.28318